LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py

```diff
@@ -148,10 +168,29 @@ def generate_MSI_token():
     try:
         data = None
         while retries <= max_retries:
-            msiauthurl = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://ingestion.monitor.azure.com/"
-            req = urllib2.Request(msiauthurl, headers={'Metadata':'true', 'Content-Type':'application/json'})
-            res = urllib2.urlopen(req)
-            data = json.loads(res.read())
+            arc_endpoint = get_arc_endpoint()
+            try:
+                msiauthurl = arc_endpoint + "/metadata/identity/oauth2/token?api-version=2019-11-01&resource=https://management.azure.com/"
+                req = urllib2.Request(msiauthurl, headers={'Metadata':'true'})
```

Comment:
If it’s expected to fail without token, then why make this call without token

Reply:
This is by design from the arc team. The tokens don't exist by default, we only get them after the first call fails, then we use that token to again make the call with auth.